ember-addon-main.js

```diff
@@ -106,7 +107,10 @@ module.exports = {
         ast: pluginInfo.plugins
       },
 
-      pluginCacheKey: pluginInfo.cacheKeys
+      pluginCacheKey: pluginInfo.cacheKeys,
+
+      templateCacheKey: pluginInfo.templateCacheKeys &&
+        templateCacheKey.bind(null, pluginInfo.templateCacheKeys)
```

Comment:
Huh. The ast plugins are functions. Is that why they are registered into a global scope? If so, how does https://github.com/ember-cli/ember-cli-htmlbars/pull/130 handle this? Can the same mechanism be used here?

In a separate thread, I'm currently working on a patch to https://github.com/stefanpenner/broccoli-persistent-filter that enables dependency tracking. Basically there's an instance assigned to the broccoli plugin that the implementation of `processString()` can use to register files that the current file depends on after its done doing its processing. Using that metadata, the persistent filter can invalidate the persistent cache as well as invalidate files in the tree during a rebuild. I suspect that the dependency tracker approach may be more friendly to the parallel transpilation feature.

Can you please let me know what you think is a good way to connect these dots that will be compatible with parallel transpilation? Should the cacheKey and/or dependencies be provided on the AST plugin object instead of on the wrapper API? I've been assuming the wrapper was the right place because it keeps the concerns of the separate projects distinct, but at least for my code, the ast plugin itself is the natural place to write this logic. I was planning to use a closure on my end to delegate from the wrapper to the plugin.